### PR TITLE
[JENKINS-73813] Show a notification when scheduling a build fails

### PR DESCRIPTION
**Pull request by mawinter69**
When a build is triggered via the UI (either on the job page or via the buildbuttoncolumn in a view) and the crumb is no longer valid, the tooltip was shown that the build was scheduled, although nothing happened.
This change will show a notification when scheduling failed.

image

See [JENKINS-73813](https://issues.jenkins.io/browse/JENKINS-73813).

### Testing done
Manual testing
- setup jenkins with users
- create a job without parameters
- go to dashboard
- in a second window logout the user
- click on schedule build -> notification is shown

### Proposed changelog entries

- Show a notification when scheduling a build fails

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by mawinter69** [comment]
I've chosen to use a notificationBar in case of an error as the hoverNotification doesn't allow to change color.
One idea might be to use a notification also for the success case.

**Comment by janfaracik** [comment]
> I've chosen to use a notificationBar in case of an error as the hoverNotification doesn't allow to change color. One idea might be to use a notification also for the success case.

I think it'd make sense to standardise on one - +1 on the notification.

File: core/src/main/resources/hudson/views/BuildButtonColumn/icon.js
Comment (janfaracik) [inline core/src/main/resources/hudson/views/BuildButtonColumn/icon.js:22]:
```suggestion
      });
```

**Review by janfaracik — APPROVED** [review]
LGTM - thanks :)

File: core/src/main/resources/hudson/views/BuildButtonColumn/column.properties
Comment (janfaracik) [inline core/src/main/resources/hudson/views/BuildButtonColumn/column.properties:26]:
```suggestion
Task_schedule_failed=Failed to schedule build for {0}
```

For consistency with the 'Build scheduled' content

**Comment by janfaracik** [comment]
Couple linting issues.

**Comment by daniel-beck** [comment]
This does not address the issue with Build Now from context menus (e.g., breadcrumb bar). Is that deliberate or just an oversight?

**Comment by mawinter69** [comment]
> This does not address the issue with Build Now from context menus (e.g., breadcrumb bar). Is that deliberate or just an oversight?

ah yes, that I overlooked

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by daniel-beck — APPROVED** [review]

**Pull request merged**
